features/lorawan/lorastack/mac/LoRaMacCrypto.h

```diff
@@ -26,97 +26,150 @@ SPDX-License-Identifier: BSD-3-Clause
 #ifndef MBED_LORAWAN_MAC_LORAMAC_CRYPTO_H__
 #define MBED_LORAWAN_MAC_LORAMAC_CRYPTO_H__
 
-/**
- * Computes the LoRaMAC frame MIC field
- *
- * @param [in]  buffer          - Data buffer
- * @param [in]  size            - Data buffer size
- * @param [in]  key             - AES key to be used
- * @param [in]  address         - Frame address
- * @param [in]  dir             - Frame direction [0: uplink, 1: downlink]
- * @param [in]  seq_counter     - Frame sequence counter
- * @param [out] mic             - Computed MIC field
- *
- * @return                        0 if successful, or a cipher specific error code
- */
-int compute_mic(const uint8_t *buffer, uint16_t size, const uint8_t *key,
-                      uint32_t address, uint8_t dir, uint32_t seq_counter,
-                      uint32_t *mic);
+#include "mbedtls/aes.h"
+#include "mbedtls/cmac.h"
 
 /**
- * Performs payload encryption
- *
- * @param [in]  buffer          - Data buffer
- * @param [in]  size            - Data buffer size
- * @param [in]  key             - AES key to be used
- * @param [in]  address         - Frame address
- * @param [in]  dir             - Frame direction [0: uplink, 1: downlink]
- * @param [in]  seq_counter     - Frame sequence counter
- * @param [out] enc_buffer      - Encrypted buffer
- *
- * @return                        0 if successful, or a cipher specific error code
+ * CMAC/AES Message Integrity Code (MIC) Block B0 size
  */
-int encrypt_payload(const uint8_t *buffer, uint16_t size, const uint8_t *key,
-                    uint32_t address, uint8_t dir, uint32_t seq_counter,
-                    uint8_t *enc_buffer);
+#define LORAMAC_MIC_BLOCK_B0_SIZE       16
 
-/**
- * Performs payload decryption
- *
- * @param [in]  buffer          - Data buffer
- * @param [in]  size            - Data buffer size
- * @param [in]  key             - AES key to be used
- * @param [in]  address         - Frame address
- * @param [in]  dir             - Frame direction [0: uplink, 1: downlink]
- * @param [in]  seq_counter     - Frame sequence counter
- * @param [out] dec_buffer      - Decrypted buffer
- *
- * @return                        0 if successful, or a cipher specific error code
- */
-int decrypt_payload(const uint8_t *buffer, uint16_t size, const uint8_t *key,
-                    uint32_t address, uint8_t dir, uint32_t seq_counter,
-                    uint8_t *dec_buffer);
+#define LORAMAC_AES_CMAC_KEY_LENGTH     16
```

Comment:
On a similar note, this wouldn't be needed if you passed the key length as a parameter to compute_mic, and that's `sizeof nwk_skey`. Would actually be a bit tighter. Or the key could actually be a proper type - a struct with a 16-byte array in it, so you can use `sizeof` inside `compute_mic`.